server/src/test/java/io/unitycatalog/server/base/model/BaseModelCRUDTest.java

```diff
@@ -0,0 +1,152 @@
+  protected SchemaOperations schemaOperations;
+  protected ModelOperations modelOperations;
+
+  protected abstract SchemaOperations createSchemaOperations(ServerConfig serverConfig);
+
+  protected abstract ModelOperations createModelOperations(ServerConfig serverConfig);
+
+  @BeforeEach
+  @Override
+  public void setUp() {
+    super.setUp();
+    schemaOperations = createSchemaOperations(serverConfig);
+    modelOperations = createModelOperations(serverConfig);
+  }
+
+  protected void createCommonResources() throws ApiException {
+    CreateCatalog createCatalog = new CreateCatalog().name(CATALOG_NAME).comment(COMMENT);
+    catalogOperations.createCatalog(createCatalog);
+    schemaOperations.createSchema(new CreateSchema().name(SCHEMA_NAME).catalogName(CATALOG_NAME));
+  }
+
+  @Test
+  public void testModelCRUD() throws ApiException {
+    // Model doesn't exist
+    assertThatThrownBy(() -> modelOperations.getRegisteredModel(MODEL_FULL_NAME))
+        .isInstanceOf(Exception.class);
+
+    // Model creation fails if missing catalog/schema
+    CreateRegisteredModel createRm =
+        new CreateRegisteredModel()
+            .name(MODEL_NAME)
+            .catalogName(CATALOG_NAME)
+            .schemaName(SCHEMA_NAME)
+            .comment(COMMENT);
+    assertThatThrownBy(() -> modelOperations.createRegisteredModel(createRm))
+        .isInstanceOf(Exception.class);
+
+    // Create a catalog/schema
+    createCommonResources();
+
+    // Create a registered model
+    System.out.println("Testing create registered model...");
+    RegisteredModelInfo rmInfo = modelOperations.createRegisteredModel(createRm);
+    assertThat(rmInfo.getName()).isEqualTo(createRm.getName());
+    assertThat(rmInfo.getCatalogName()).isEqualTo(createRm.getCatalogName());
+    assertThat(rmInfo.getSchemaName()).isEqualTo(createRm.getSchemaName());
+    assertThat(rmInfo.getFullName()).isEqualTo(MODEL_FULL_NAME);
+    assertThat(rmInfo.getComment()).isEqualTo(createRm.getComment());
+    assertThat(rmInfo.getCreatedAt()).isNotNull();
+    assertThat(rmInfo.getUpdatedAt()).isNotNull();
+    assertThat(rmInfo.getModelId()).isNotNull();
+    assertThat(rmInfo.getStorageLocation()).isNotNull();
+
+    // List registered models
+    System.out.println("Testing list registered models..");
+    Iterable<RegisteredModelInfo> modelList =
+        modelOperations.listRegisteredModels(CATALOG_NAME, SCHEMA_NAME);
+    assertThat(modelList).contains(rmInfo);
+
+    // Get registered model
+    System.out.println("Testing get registered model..");
+    RegisteredModelInfo retrievedRmInfo = modelOperations.getRegisteredModel(MODEL_FULL_NAME);
+    assertThat(retrievedRmInfo).isEqualTo(rmInfo);
+
+    // Calling update model with nothing to update should throw an exception
+    System.out.println("Testing updating registered model with nothing to update..");
+    UpdateRegisteredModel emptyUpdateRegisteredModel = new UpdateRegisteredModel();
+    assertThatThrownBy(
```

Comment:
I know we have not done this before but maybe checking of specific error messages might be a good practice going forward.

Reply:
I don't think matching against error message is a good idea since it would make the tests very brittle to error message changes.  I think we do error code checking in the DB UC tests, which I can add in.

Reply:
+1, we can do an error handling overhaul in another PR.